bundles/org.openhab.binding.siemenshvac/src/main/java/org/openhab/binding/siemenshvac/internal/metadata/SiemensHvacMetadataRegistryImpl.java

```diff
@@ -0,0 +1,1270 @@
+import org.openhab.binding.siemenshvac.internal.converter.TypeConverter;
+import org.openhab.binding.siemenshvac.internal.handler.SiemensHvacBridgeConfig;
+import org.openhab.binding.siemenshvac.internal.network.SiemensHvacCallback;
+import org.openhab.binding.siemenshvac.internal.network.SiemensHvacConnector;
+import org.openhab.binding.siemenshvac.internal.type.SiemensHvacChannelGroupTypeProvider;
+import org.openhab.binding.siemenshvac.internal.type.SiemensHvacChannelTypeProvider;
+import org.openhab.binding.siemenshvac.internal.type.SiemensHvacConfigDescriptionProvider;
+import org.openhab.binding.siemenshvac.internal.type.SiemensHvacException;
+import org.openhab.binding.siemenshvac.internal.type.SiemensHvacThingTypeProvider;
+import org.openhab.binding.siemenshvac.internal.type.UidUtils;
+import org.openhab.core.OpenHAB;
+import org.openhab.core.config.core.ConfigDescriptionBuilder;
+import org.openhab.core.config.core.ConfigDescriptionParameter;
+import org.openhab.core.config.core.ConfigDescriptionParameterGroup;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.type.ChannelDefinition;
+import org.openhab.core.thing.type.ChannelDefinitionBuilder;
+import org.openhab.core.thing.type.ChannelGroupDefinition;
+import org.openhab.core.thing.type.ChannelGroupType;
+import org.openhab.core.thing.type.ChannelGroupTypeBuilder;
+import org.openhab.core.thing.type.ChannelGroupTypeUID;
+import org.openhab.core.thing.type.ChannelType;
+import org.openhab.core.thing.type.ChannelTypeBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.thing.type.StateChannelTypeBuilder;
+import org.openhab.core.thing.type.ThingType;
+import org.openhab.core.thing.type.ThingTypeBuilder;
+import org.openhab.core.types.StateDescriptionFragmentBuilder;
+import org.openhab.core.types.StateOption;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+
+/**
+ *
+ * @author Laurent Arnal - Initial contribution
+ */
+@NonNullByDefault
+@Component(immediate = true)
+public class SiemensHvacMetadataRegistryImpl implements SiemensHvacMetadataRegistry {
+
+    private final Logger logger = LoggerFactory.getLogger(SiemensHvacMetadataRegistryImpl.class);
+
+    // A map contains data point config read from Api and/or WebPages
+    private Map<String, SiemensHvacMetadata> dptMap = new Hashtable<String, SiemensHvacMetadata>();
+    private @Nullable SiemensHvacMetadata root = null;
+    private @Nullable ArrayList<SiemensHvacMetadataDevice> devices = null;
+
+    private static final String JSON_DIR = OpenHAB.getUserDataFolder() + File.separatorChar + "jsondb";
```

Comment:
Yes, why not.
I wait for your confirmation if it's the good way to do it.

Reply:
For SiemensHvacRequestListener, review the exception throw clause on method onComplete and onError, and change to catch only SiemensHvacException.

Fix on next commit

Reply:
Storing the data inside `userdata/siemenshvac` should be safe as other bindings do this as well. Examples:

https://github.com/openhab/openhab-addons/blob/2645a264433c683224bd5da8148e06a7cdbb54ed/bundles/org.openhab.binding.androidtv/src/main/java/org/openhab/binding/androidtv/internal/protocol/googletv/GoogleTVConnectionManager.java#L453

https://github.com/openhab/openhab-addons/blob/2645a264433c683224bd5da8148e06a7cdbb54ed/bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/MiIoBindingConstants.java#L133-L134

However, if you prefer the `jsondb` subdirectory, we would have to await input from @openhab/add-ons-maintainers since I don't know if there are rules against that.

Reply:
Ok, I change the directory to be userdata/siemenshvac.
fix on next commit